test/box-luatest/error_subsystem_improvements_test.lua

```diff
@@ -169,3 +169,28 @@ g.test_unpack = function()
     t.assert_equals(u.custom_type, nil)
     t.assert_equals(type(u.code), 'number')
 end
+
+-- Test ClientError arguments become payload fields (gh-9109/gh-9108).
+g.test_client_error_creation = function()
```

Comment:
Instead of converting one error for tests, I think we'd better just write some C unit tests checking that the new macros work as expected (i.e. set all payload fields as specified). Also, it's better to introduce all possible payload field types in this commit.

Reply:
Added some test error as discussed f2f.